### PR TITLE
Bln 1

### PR DESCRIPTION
**Pull request by gerardhros**
## Added
* SOMERS calculation of emissions in peat soils, function `bln_clim_somers` and associated test function
* internal package table for SOMERS meta-model `bln_somers`

## Updated
* `ppr_tables` in dev to prepare internal package table `bln_somers`

### CODE REVIEW

**Comment by dendrophil** [comment]
I have placed a few commentsin R/bln_clim_somers. No other comments.

**Comment by gerardhros** [comment]
@dendrophil , did you add them as review comments or pushed them to this repo?

**Review by dendrophil — COMMENTED** [review]
See my comments

File: R/bln_clim_somers.R [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  # Check inputs
+  arg.length <- max(length(B_SOILTYPE_AGR),length(A_SOM_LOI), length(B_SOMERS_BC),
+                    length(B_DRAIN_SP), length(B_DRAIN_WP))
+  checkmate::assert_character(B_SOILTYPE_AGR, any.missing = FALSE, min.len = 1, len = arg.length)
+  checkmate::assert_subset(B_SOILTYPE_AGR, choices = BLN::bln_soiltype[bln_country=='NL',bln_soil_cat1], empty.ok = FALSE)
+  checkmate::assert_numeric(A_SOM_LOI, lower = 0.1, upper = 100, any.missing = FALSE, min.len = 1, len = arg.length)
+  checkmate::assert_integerish(B_SOMERS_BC, min.len = 1, len = arg.length)
+  checkmate::assert_subset(B_SOMERS_BC, choices = c(NA,unique(BLN::bln_somers$b_somers_bc)))
+  checkmate::assert_numeric(B_DRAIN_SP, lower = 0, upper = 1.2, min.len = 1, len = arg.length)
+  checkmate::assert_numeric(B_DRAIN_WP, lower = -0.2, upper = 1.2, min.len = 1, len = arg.length)
```
[the remark this answers is not in the document]
Comment (gerardhros) [reply R/bln_clim_somers.R:33]: B_DRAIN_WP and B_DRAIN_SP are both defined as "ontwatering depth". So, the WP is expressed in absolute numbers here.

File: R/bln_clim_somers.R [hunk 2]
```diff
@@ -0,0 +1,97 @@
+#' Function to evaluate the carbon saturation via SOMERS simulation on peat soils
+#'
+#' @param ID (character) A field id
+#' @param B_SOILTYPE_AGR (character) The agricultural type of soil
+#' @param A_SOM_LOI (numeric) value for the soil organic matter content of the soil
+#' @param B_SOMERS_BC (integer) The base combination of SOMERS peat soil classification (varies between 1 and 290)
+#' @param B_DRAIN_SP (numeric) the drooglegging of a field in summer (difference field height and ditch level, in meters)
+#' @param B_DRAIN_WP (numeric) the drooglegging of a field in winter (difference field height and ditch level, in meters)
+#' @param B_DRAIN_SP_CHANGE (numeric) the decrease in drooglegging of a field in summer (in meters). Allowed decrease varies from 0 to 0.5m.
```
[the remark this answers is not in the document]
Comment (gerardhros) [reply R/bln_clim_somers.R:9]: for each bc combination, the summer drainage level (ontwatering) varies from 0.2 to 1.2 m. this variable is a predefined change that due to an increase in water level the drainage level can be up to 50cm higher than the original one. higher values are not likely to occur. See line 66 how this change is added on top of the actual drainage level in summer. the pmin and pmax ensures that the new level is always between 0.2 and 1.2m.

[the remark this answers is not in the document]
Comment (dendrophil) [reply R/bln_clim_somers.R:9]: Okay, thanks for the explanation!

**Pull request merged**